File: Beecrowd/Beginner./1021.cpp
#include <iostream>
#include <cmath>
using namespace std;

int main() {
    double N;
    cin >> N;

    int notes = N;
    int coins = round((N - notes) * 100);

    cout << "NOTAS:" << endl;
    cout << notes / 100 << " nota(s) de R$ 100.00" << endl;
    notes %= 100;
    cout << notes / 50 << " nota(s) de R$ 50.00" << endl;
    notes %= 50;
    cout << notes / 20 << " nota(s) de R$ 20.00" << endl;
    notes %= 20;
    cout << notes / 10 << " nota(s) de R$ 10.00" << endl;
    notes %= 10;
    cout << notes / 5 << " nota(s) de R$ 5.00" << endl;
    notes %= 5;
    cout << notes / 2 << " nota(s) de R$ 2.00" << endl;
    notes %= 2;

    coins += notes * 100;

    cout << "MOEDAS:" << endl;
    cout << coins / 100 << " moeda(s) de R$ 1.00" << endl;
    coins %= 100;
    cout << coins / 50 << " moeda(s) de R$ 0.50" << endl;
    coins %= 50;
    cout << coins / 25 << " moeda(s) de R$ 0.25" << endl;
    coins %= 25;
    cout << coins / 10 << " moeda(s) de R$ 0.10" << endl;
    coins %= 10;
    cout << coins / 5 << " moeda(s) de R$ 0.05" << endl;
    coins %= 5;
    cout << coins / 1 << " moeda(s) de R$ 0.01" << endl;

    return 0;
}

File: Beecrowd/1. Beginner/1021/1021.java
import java.util.Scanner;

public class Main {
    public static void main(String[] args){
        Scanner input = new Scanner(System.in);

        double amount = input.nextDouble();

        int notes = (int)amount;
        int coins =(int) ((amount - notes)*100);
        
        System.out.println("NOTAS:");
        System.out.printf("%d nota(s) de R$ 100.00\n", notes/100);
        notes %= 100;

        System.out.printf("%d nota(s) de R$ 50.00\n", notes/50);
        notes %= 50;

        System.out.printf("%d nota(s) de R$ 20.00\n", notes/20);
        notes %= 20;

        System.out.printf("%d nota(s) de R$ 10.00\n", notes/10);
        notes %= 10;

        System.out.printf("%d nota(s) de R$ 5.00\n", notes/5);
        notes %= 5;
        
        System.out.printf("%d nota(s) de R$ 2.00\n", notes/2);
        notes %= 2;

        coins += notes * 100;

        System.out.println("MOEDAS:");

        System.out.printf("%d moeda(s) de R$ 1.00\n", coins/100);
        coins %= 100;

        System.out.printf("%d moeda(s) de R$ 0.50\n", coins/50);
        coins %= 50;

        System.out.printf("%d moeda(s) de R$ 0.25\n", coins/25);
        coins %= 25;
        
        System.out.printf("%d moeda(s) de R$ 0.10\n", coins/10);
        coins %= 10;
        
        System.out.printf("%d moeda(s) de R$ 0.05\n", coins/5);
        coins %= 5;
        
        System.out.printf("%d moeda(s) de R$ 0.01\n", coins/1);
        
        input.close();
    }
}